doc/develop/test/twister_statuses.rst

```diff
@@ -0,0 +1,215 @@
+.. _twister_statuses:
+
+Twister Status
+##############
+
+What is a Twister Status?
+=========================
+
+Twister Status formulates the current state of
+
+- ``Harness``
+- ``TestCase``
+- ``TestSuite``
+- ``TestInstance``
+
+in a comprehensive and easy-to understand way.
+In practice, most users will be interested in Statuses
+of Instances and Cases after the conclusion of their Twister runs.
```

Comment:
```suggestion
```
IMO its better to remove this from here to allow more clarification of what 'instance' is from user's perspective later in the 'nomenclature' section below.